client/app/views/home_application.coffee

```diff
@@ -59,10 +59,11 @@ module.exports = class ApplicationRow extends BaseView
                     extension = 'png'
                     @icon.removeClass 'svg'
 
-                src = "api/applications/#{app.id}.#{extension}"
+                src = "api/applications/#{@model.id}.#{extension}"
                 @icon.attr 'src', src
                 @icon.show()
                 @icon.removeClass 'stopped'
+                $("##{@model.get 'slug'}-frame").remove()
```

Comment:
Are you sure that the iFrame is recreated when the app is installed and started again?